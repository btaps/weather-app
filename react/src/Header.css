.header {
  text-align: center;
  /* font-weight: 800; */
  margin-top: 2rem;
}

.header__title {
  letter-spacing: 5px;
  font-weight: 100;
  margin: 0;
  font-size: 4rem;
  text-transform: uppercase;
}

.header__info {
  font-size: 1.5rem;
}
