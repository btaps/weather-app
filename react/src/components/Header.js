import React from "react";
import "./Header.css";

function Header() {
  return (
    <header className="header">
      <h1 className="header__title">Weather App</h1>
      <p className="header__info">Search for a city or locate yourself!</p>
    </header>
  );
}

export default Header;
