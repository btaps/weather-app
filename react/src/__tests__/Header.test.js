import { render, screen, cleanup } from "@testing-library/react";
import Header from "../components/Header.js";

import renderer from "react-test-renderer";

afterEach(cleanup);

test("conatins title", () => {
  render(<Header />);
  const title = screen.getByText(/Weather App/i);
  expect(title).toBeInTheDocument();
});

test("contains sub-title", () => {
  render(<Header />);
  const subTitle = screen.getByText(/Search for a city or locate yourself!/i);
  expect(subTitle).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
