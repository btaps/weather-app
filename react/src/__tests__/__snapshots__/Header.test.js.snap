// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Weather App
  </h1>
  <p
    className="header__info"
  >
    Search for a city or locate yourself!
  </p>
</header>
`;
